include/swift/Remote/MetadataReader.h

```diff
@@ -193,8 +195,18 @@ class MetadataReader {
   /// amounts of data when we encounter corrupt values for sizes/counts.
   static const uint64_t MaxMetadataSize = 1048576; // 1MB
 
-  /// A cache of built types, keyed by the address of the type.
-  std::unordered_map<StoredPointer, BuiltType> TypeCache;
+  /// Define a has function for a std::pair<StoredPointer, bool>
+  struct TypeCacheKeyHash {
+    std::size_t operator()(const std::pair<StoredPointer, bool> &x) const {
+      return llvm::hash_combine(x.first, x.second);
+    }
+  };
+
+  /// A cache of built types, keyed by the address of the type and whether the
+  /// request ignored articial superclasses or not.
+  std::unordered_map<std::pair<StoredPointer, bool>, BuiltType,
+                     TypeCacheKeyHash>
```

Comment:
Just curious: If we are using the LLVM libraries anyway, why not use a DenseMap?

Reply:
Looks like we can, I'll change this in a separate patch.